FirebaseStorage/Tests/Integration/FIRStorageIntegrationTests.m

```diff
@@ -413,6 +413,32 @@ - (void)testGetDataInBackgroundQueue {
   [self waitForExpectations];
 }
 
+- (void)testGetDataWithCustomCallbackQueue {
+  XCTestExpectation *expectation =
+      [self expectationWithDescription:@"testUnauthenticatedGetDataInCustomCallbackQueue"];
+
+  dispatch_queue_t callbackQueue = dispatch_queue_create("customCallbackQueue", NULL);
+  _storage.callbackQueue = callbackQueue;
+
+  FIRStorageReference *ref = [self.storage referenceWithPath:@"ios/public/1mb"];
+  [ref dataWithMaxSize:1 * 1024 * 1024
+            completion:^(NSData *data, NSError *error) {
+              if (@available(iOS 10.0, macOS 10.12, *)) {
+                dispatch_assert_queue(callbackQueue);
```

Comment:
I would suggest to be consistent between ObjC and Swift tests with either option:
1.  Use the queue specifics for both
2. Use `dispatch_assert_queue` and [dispatchPrecondition](https://developer.apple.com/documentation/dispatch/1780605-dispatchprecondition)
3. Have separate tests with queue specifics for old versions (to be deleted once supported versions updated) and `dispatch_assert_queue` based for modern versions